src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java

```diff
@@ -4,22 +4,29 @@
 import org.slf4j.LoggerFactory;
 import org.yb.cdc.CdcService;
 
+import java.math.BigInteger;
 import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.PriorityBlockingQueue;
 
+/**
+ * @author Rajat Venkatesh
+ */
 public class Merger {
     private static final Logger LOGGER = LoggerFactory.getLogger(Merger.class);
-    private final PriorityQueue<Message> queue = new PriorityQueue<>();
-    private final Map<String, List<Message>> mergeSlots = new HashMap<>();
-    private final Map<String, Long> tabletSafeTime = new HashMap<>();
+    private final PriorityBlockingQueue<Message> queue = new PriorityBlockingQueue<>();
```

Comment:
Can you move the data structures back to non-concurrent ones? There is no point in taking locks at function and data structure level.